Eatery/Controllers/Eateries/Campus/CampusEateriesSearchViewController.swift

```diff
@@ -6,22 +6,352 @@
+    let subtitle: String
+    let isFavorite: Bool?
+
+}
+
+private class SearchResultsManager {
+
+    var eateries: [CampusEatery] = []
+    
+    func searchResult(searchText: String) -> [SearchResult] {
+        let searchResults = eaterySearchResults(searchText: searchText)
+            + menuItemSearchResults(searchText: searchText)
+            + areaSearchResults(searchText: searchText)
+        
+        return searchResults
+    }
+
+    private func eaterySearchResults(searchText: String) -> [SearchResult] {
+        eateries.filter { eatery in
+            matches(eatery.name, searchText) || matches(eatery.displayName, searchText)
+        }.map { eatery in
+            let subtitle: String
+            switch eatery.eateryType {
+            case .dining: subtitle = "Dining Hall"
+            default: subtitle = "Café"
+            }
+
+            return SearchResult(
+                source: .eatery(eatery),
+                title: eatery.displayName,
+                subtitle: subtitle,
+                isFavorite: eatery.isFavorite)
+        }
+    }
+
+    private func menuItemSearchResults(searchText: String) -> [SearchResult] {
+        var searchResults: [SearchResult] = []
+
+        for eatery in eateries {
+            var items = Set(eatery.diningItems(onDayOf: Date()))
+            if let activeEvent = eatery.activeEvent(atExactly: Date()) {
+                items.formUnion(activeEvent.menu.data.values.flatMap { $0 })
+            }
+
+            let filteredItems = items
+                .filter { matches($0.name, searchText) }
+                .sorted { $0.name < $1.name }
+
+            searchResults += filteredItems.map {
+                SearchResult(
+                    source: .menuItem(eatery, $0),
+                    title: $0.name,
+                    subtitle: eatery.displayName,
+                    isFavorite: false)
+            }
+        }
+
+        return searchResults
+    }
+
+    private func areaSearchResults(searchText: String) -> [SearchResult] {
+        Area.allCases
+            .filter { area in matches(area.description, searchText) }
+            .map { area in
+                SearchResult(
+                    source: .area(area),
+                    title: area.description,
+                    subtitle: "Location",
+                    isFavorite: nil)
+        }
+    }
+
+    private func matches(_ textToSearch: String, _ searchText: String) -> Bool {
+        return textToSearch.localizedCaseInsensitiveContains(searchText)
+    }
+
+}
+
 class CampusEateriesSearchViewController: UIViewController {
 
+    // Model
+
+    private var mode: Mode = .recentSearches
+    private let searchResults = SearchResultsManager()
+
+    private var displayedRecentSearches: [RecentSearch] = []
+    private var displayedSearchResults: [SearchResult] = []
+
+    // Views
+
+    private var titleLabel: UILabel!
+    private var clearButton: UIButton!
+
+    private var tableView: UITableView!
```

Comment:
You can set all of these views right here. That way it's more concise, you don't need `!`, and you can use `let`.
```
private let titleLabel = UILabel()
private let clearButton = UIButton(type: .system)
private let tableView = UITableView()
```